pkg/binding/buffering/copy_message.go

```diff
@@ -32,7 +32,7 @@ func CopyMessage(ctx context.Context, m binding.Message, transformers binding.Tr
 		if err != nil {
 			return nil, err
 		}
-		return binding.EventMessage(e), nil
+		return binding.EventMessage(*e), nil
```

Comment:
This provokes a copy, you should do `(*binding.EventMessage)(e)` to just cast it as a pointer